applications/StructuralMechanicsApplication/custom_advanced_constitutive/wrinkling_linear_2d_law.cpp

```diff
@@ -0,0 +1,283 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Klaus B. Sautter
+//
+// System includes
+#include <iostream>
+#include <set>
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "custom_advanced_constitutive/wrinkling_linear_2d_law.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+/******************************CONSTRUCTOR******************************************/
+/***********************************************************************************/
+
+WrinklingLinear2DLaw::WrinklingLinear2DLaw()
+    : ConstitutiveLaw()
+{
+}
+
+/******************************COPY CONSTRUCTOR*************************************/
+/***********************************************************************************/
+
+WrinklingLinear2DLaw::WrinklingLinear2DLaw(const WrinklingLinear2DLaw& rOther)
+    : ConstitutiveLaw(rOther),
+      mpConstitutiveLaw(rOther.mpConstitutiveLaw)
+{
+}
+
+/********************************CLONE**********************************************/
+/***********************************************************************************/
+
+ConstitutiveLaw::Pointer WrinklingLinear2DLaw::Clone() const
+{
+    return Kratos::make_shared<WrinklingLinear2DLaw>(*this);
+}
+
+/*******************************CONSTRUCTOR*****************************************/
+/***********************************************************************************/
+
+ConstitutiveLaw::Pointer WrinklingLinear2DLaw::Create(Kratos::Parameters NewParameters) const
+{
+    return Kratos::make_shared<WrinklingLinear2DLaw>();
+}
+
+//*******************************DESTRUCTOR*******************************************
+/***********************************************************************************/
+
+WrinklingLinear2DLaw::~WrinklingLinear2DLaw()
+{
+};
+
+
```

Comment:
there are many empty lines in this file, I would separate with one empty line btw functions

Reply:
>  I would separate with one empty line btw functions
?

Reply:
you often have 3 or more empty lines btw functions
=> reduce to 1